doc/development/style_guide.md

```diff
@@ -0,0 +1,870 @@
+
+```text
+File '/path/to/file.txt' not found.
+```
+
+First letter should be capitalized.
+
+Avoid contractions (cannot instead of can't).
+
+Be consistent with periods. Complete sentences or all parts of a message with
+multiple sentences should end with periods. Short phrases should not.
+Punctuated events, such as errors, deserve a period, e.g., _"Operation
+complete."_ Phrases which imply ongoing action should have an ellipse, e.g.,
+ _"Reading raster map..."_.
+
+### Developing Python scripts
+
+#### Import Python Scripting Library
+
+```python
+import grass.script as gs
+
+gs.run_command(...)
+```
+
+#### String formatting
+
+User messages should be translatable and for formatting, use `str.format()`, not
+f-strings:
+
+```python
+gs.warning(_("Raster map <{}> not found.").format(input_map))
+```
+
+For strings that are not translatable, use f-strings:
+
+```python
+r_mapcalc_expression = f"{output_map} = {input_map} * 3"
+```
+
+#### Temporary files
+
+To create a temporary file, use `NamedTemporaryFile` with a context manager. In
+this example, we open a temporary file for writing, write something and then we
+can use it in another tool. Once we do not need it anymore, we need to delete it
+ourselves.
+
+```python
+import tempfile
+
+with tempfile.NamedTemporaryFile(mode="w", delete=False) as tmp_file:
+    file_path = tmp_file.name
+    tmp_file.write(...)
+
+gs.try_remove(file_path)
+```
+
+#### Changing computational region
+
+If a tool needs to change the computational region for part of the computation,
+temporary region in Python API is the simplest way to do it:
+
+```python
+gs.use_temp_region()  # From now on, use a separate region in the script.
+# Set the computational region with g.region as needed.
+grass.run_command('g.region', raster='input')
+gs.del_temp_region()
+# Original region applies now.
+```
+
+This makes any changes done in the tool local for the tool without influencing
+other tools running in the same session.
+
+If you need even more control, use the GRASS_REGION environment variable which
+is passed to subprocesses. Python API has functions which help with the setup:
+
+```python
+os.environ["GRASS_REGION"] = gs.region_env(raster=input_raster)
+```
+
+If different subprocesses need different regions, use different environments:
+
+```python
+env = os.environ.copy()
+env["GRASS_REGION"] = gs.region_env(raster=input_raster)
+gs.run_command("r.slope.aspect", elevation=input_raster, slope=slope, env=env)
+```
+
+This approach makes the computational region completely safe for parallel
+processes as no region-related files are modified.
+
+#### Temporary Maps
+
+Using temporary maps is preferred over using temporary mapsets. This follows the
+rule that writing should be done only to the current mapset. Some users may have
+write permissions only for their mapsets, but not for creating other mapsets.
+
+The following script creates a temporary name using `gs.append_node_pid`` which
```

Comment:
```suggestion
The following script creates a temporary name using `gs.append_node_pid` which
```

Was the extra backtick intended (the PR syntax formatting was weird on my phone, maybe that was why)

Reply:
No, my editor is (partially) to blame